### ISSUE TITLE
'InstantXFluxIPAdapterModel' object has no attribute 'update_ip_adapter_advanced'

### ISSUE THREAD

**Issue by gioferreira** [Shakker-Labs/ComfyUI-IPAdapter-Flux#96]
# ComfyUI Error Report
## Error Details
- **Node ID:** 876
- **Node Type:** ApplyIPAdapterFluxAdvanced
- **Exception Type:** AttributeError
- **Exception Message:** 'InstantXFluxIPAdapterModel' object has no attribute 'update_ip_adapter_advanced'
## Stack Trace
```
  File "D:\ComfyUI-Seara\execution.py", line 349, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "D:\ComfyUI-Seara\execution.py", line 224, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "D:\ComfyUI-Seara\execution.py", line 196, in _map_node_over_list
    process_inputs(input_dict, i)

  File "D:\ComfyUI-Seara\execution.py", line 185, in process_inputs
    results.append(getattr(obj, func)(**inputs))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "D:\ComfyUI-Seara\custom_nodes\ComfyUI-IPAdapter-Flux\ipadapter_flux_advanced.py", line 178, in apply_ipadapter_flux_advanced
    ipadapter_flux.update_ip_adapter_advanced(model.model, (weight_start, weight_end, steps), (start_percent, end_percent))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```
## System Information
- **ComfyUI Version:** 0.3.39
- **Arguments:** main.py --use-sage-attention --auto-launch
- **OS:** nt
- **Python Version:** 3.12.8 (tags/v3.12.8:2dc476b, Dec  3 2024, 19:30:04) [MSC v.1942 64 bit (AMD64)]
- **Embedded Python:** false
- **PyTorch Version:** 2.7.1+cu128
## Devices

- **Name:** cuda:0 NVIDIA GeForce RTX 4090 : cudaMallocAsync
  - **Type:** cuda
  - **VRAM Total:** 25756565504
  - **VRAM Free:** 2833324906
  - **Torch VRAM Total:** 21206401024
  - **Torch VRAM Free:** 33626986

## Additional Context
It only happens with the advanced node

**Comment by philipy1219** [comment]
Use IPAdapterFluxLoaderAdvanced With ApplyIPAdapterFluxAdvanced

**Comment by kenic123** [comment]
这个地方居然没有issue,难道没一个人用吗
将ipadapter_flux_advanced.py文件的第69行self.ip_attn_procs = self.init_ip_adapter()，改为self.ip_attn_procs = self.init_ip_adapter_advanced()可以解决报错问题

**Comment by kenic123** [comment]
而且在固定种子只改变参数的情况下第二遍执行会报没有to_k_ip错

**Comment by hjcxlp** [comment]
> 改成这样就可以了 import torch import torch.nn as nn import torch.nn.functional as F import weakref from diffusers.models.normalization import RMSNorm from einops import rearrange
> 
> class IPAFluxAttnProcessor2_0Advanced(nn.Module): _instances = weakref.WeakSet() _global_call_count = 0 _last_timestep_printed = None _first_instance_for_timestep = None # Add this line
> 
> ```
> def __init__(self, num_tokens, hidden_size, cross_attention_dim=None, scale_start=1.0, scale_end=1.0, total_steps=1, timestep_range=None):
>     super().__init__()
>     self.hidden_size = hidden_size
>     self.cross_attention_dim = cross_attention_dim
>     self.scale_start = scale_start
>     self.scale_end = scale_end
>     self.total_steps = total_steps
>     self.num_tokens = num_tokens
>     
>     self.to_k_ip = nn.Linear(cross_attention_dim or hidden_size, hidden_size, bias=False)
>     self.to_v_ip = nn.Linear(cross_attention_dim or hidden_size, hidden_size, bias=False)
>     
>     self.norm_added_k = RMSNorm(128, eps=1e-5, elementwise_affine=False)
>     self.norm_added_v = RMSNorm(128, eps=1e-5, elementwise_affine=False)
>     self.timestep_range = timestep_range
> 
>     self.seen_timesteps = set()
>     self.steps = 0
>     
>     # Add this instance to the set of instances
>     self.__class__._instances.add(self)
> 
> def clear_memory(self):
>     # 只清理缓存，不删除属性
>     self.seen_timesteps.clear()
>     self.steps = 0
> 
> @classmethod
> def reset_all_instances(cls):
>     """Reset all instances of the class"""
>     cls._global_call_count = 0
>     cls._last_timestep_printed = None
>     cls._first_instance_for_timestep = None
>     for instance in cls._instances:
>         instance.seen_timesteps.clear()
>         instance.steps = 0
> 
> def reset_steps(self):
>     """Reset the steps counter and seen timesteps for this instance."""
>     self.seen_timesteps.clear()
>     self.steps = 0
>     self.__class__._last_timestep_printed = None
> 
> def __del__(self):
>     # 只清理缓存，不删除属性
>     self.clear_memory()
>     try:
>         self.__class__._instances.remove(self)
>     except KeyError:
>         pass
>         
> def __call__(
>     self,
>     num_heads,
>     query,
>     image_emb: torch.FloatTensor,
>     t: torch.FloatTensor,
> ) -> torch.FloatTensor:
>     current_timestep = t[0].item()
>     
>     # Reset steps when starting a new sequence (timestep = 1.0)
>     if abs(current_timestep - 1.0) < 1e-6:
>         self.reset_steps()
>         
>     if self.timestep_range is not None:
>         if t[0] > self.timestep_range[0] or t[0] < self.timestep_range[1]:
>             return None
>     
>     # Only update steps and print for the first instance that sees this timestep
>     if current_timestep not in self.seen_timesteps:
>         self.seen_timesteps.add(current_timestep)
>         self.steps += 1
>         
>         # Only print if this is the first instance for this timestep
>         if self.__class__._first_instance_for_timestep is None:
>             self.__class__._first_instance_for_timestep = self
>         
>         if (self.__class__._first_instance_for_timestep == self and 
>             current_timestep != self.__class__._last_timestep_printed):
>             current_step = min(self.steps, self.total_steps)
>             if self.total_steps > 1:
>                 scale = self.scale_start + (self.scale_end - self.scale_start) * (current_step - 1) / (self.total_steps - 1)
>             else:
>                 scale = self.scale_end
>                 
>             # print(f"Timestep: {current_timestep}, Step: {self.steps}/{self.total_steps}, Weight: {scale}")
>             self.__class__._last_timestep_printed = current_timestep
>     
>     # Calculate scale for return value
>     current_step = min(self.steps, self.total_steps)
>     if self.total_steps > 1:
>         scale = self.scale_start + (self.scale_end - self.scale_start) * (current_step - 1) / (self.total_steps - 1)
>     else:
>         scale = self.scale_end
>         
>     ip_hidden_states = image_emb
>     ip_hidden_states_key_proj = self.to_k_ip(ip_hidden_states)
>     ip_hidden_states_value_proj = self.to_v_ip(ip_hidden_states)
> 
>     ip_hidden_states_key_proj = rearrange(ip_hidden_states_key_proj, 'B L (H D) -> B H L D', H=num_heads)
>     ip_hidden_states_value_proj = rearrange(ip_hidden_states_value_proj, 'B L (H D) -> B H L D', H=num_heads)
> 
>     ip_hidden_states_key_proj = self.norm_added_k(ip_hidden_states_key_proj)
>     ip_hidden_states_value_proj = self.norm_added_v(ip_hidden_states_value_proj)
> 
>     ip_hidden_states = F.scaled_dot_product_attention(query.to(image_emb.device).to(image_emb.dtype), 
>                                                     ip_hidden_states_key_proj, 
>                                                     ip_hidden_states_value_proj, 
>                                                     dropout_p=0.0, is_causal=False)
> 
>     ip_hidden_states = rearrange(ip_hidden_states, "B H L D -> B L (H D)", H=num_heads)
>     ip_hidden_states = ip_hidden_states.to(query.dtype).to(query.device)
> 
>     return scale * ip_hidden_states
> ```

替换哪部分代码？
